src/client/config.rs

```diff
@@ -0,0 +1,45 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufReader, BufWriter, Write};
+use std::path::PathBuf;
+
+#[derive(Serialize, Deserialize)]
+pub struct ClientConfig {
+    version: u64,
+    pub authentication: HashMap<String, String>,
+    //default_url: Option<Url>
+}
+
+impl ClientConfig {
+    fn default_config() -> ClientConfig {
```

Comment:
As you may already know, perfect use case for [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) :p

Reply:
Thank you and done!